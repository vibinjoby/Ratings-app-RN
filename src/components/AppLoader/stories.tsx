import React from 'react'
import { storiesOf } from '@storybook/react-native'

import AppLoader from './component'

export const Story = () => <AppLoader />

storiesOf('Components', module).add('AppLoader', () => <Story />)

export default {
  title: 'AppLoader',
}
