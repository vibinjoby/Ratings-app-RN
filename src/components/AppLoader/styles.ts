import { StyleSheet } from 'react-native'
import Colors from '../../utilities/colors'

export default StyleSheet.create({
  modalWrapper: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Colors.blackMoreOpacity,
  },
  modalContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    padding: 20,
    borderRadius: 4,
    backgroundColor: Colors.white,
  },
})
