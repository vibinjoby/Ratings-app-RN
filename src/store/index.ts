import { makeVar } from '@apollo/client'

export interface UserInfoProps {
  token?: string
  userInfo?: {
    isAdmin: boolean
    name: string
    userType: string
  }
}

const userInfoVars = makeVar<UserInfoProps>({})

export default userInfoVars
