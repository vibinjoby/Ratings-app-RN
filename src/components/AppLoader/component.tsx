import React from 'react'
import { ActivityIndicator, Modal, View } from 'react-native'
import { useSelector } from 'react-redux'

import { RootState } from '../../store'
import Colors from '../../utilities/colors'
import styles from './styles'

const AppLoader = ({ children }: React.Props<any>) => {
  const loader = useSelector((state: RootState) => state.loader)
  if (!loader?.loading) return <>{children}</>

  return (
    <>
      {children}
      <Modal animationType="none" transparent visible={loader.loading}>
        <View style={styles.modalWrapper}>
          <View style={styles.modalContainer}>
            <ActivityIndicator animating color={Colors.appOrange} size="large" />
          </View>
        </View>
      </Modal>
    </>
  )
}

export default AppLoader
