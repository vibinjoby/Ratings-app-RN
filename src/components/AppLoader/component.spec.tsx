import 'react-native'
import React from 'react'
import { render } from '@testing-library/react-native'
import { NavigationContainer } from '@react-navigation/native'
import { Provider } from 'react-redux'

import AppLoader from './component'
import { store } from '../../store'

describe('AppLoader', () => {
  it('should render', () => {
    const { toJSON } = render(
      <NavigationContainer>
        <Provider store={store}>
          <AppLoader />
        </Provider>
      </NavigationContainer>,
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
