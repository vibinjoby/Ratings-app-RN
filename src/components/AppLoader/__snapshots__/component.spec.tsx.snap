// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLoader should render 1`] = `null`;
